frontends/telegram/app/bot.py

```diff
@@ -1,23 +1,28 @@
 """
     Main Telegram bot for Subscriby.
 """
-
-from aiogram.types import ParseMode
-from aiogram.contrib.fsm_storage.memory import MemoryStorage
+from aiogram.fsm.storage.memory import MemoryStorage
 from aiogram import Dispatcher, Bot
 
 from .settings import Settings
-from .handlers import register_handlers
-
-# Initializing bot with dispatcher.
-token = Settings().subscriby_telegram_token
-bot = Bot(token=token, parse_mode=ParseMode.HTML)
-dp = Dispatcher(bot, storage=MemoryStorage())
-register_handlers(dp)
+from .middlewares.role import RoleMiddleware
+from .handlers import admin, user
 
 
 async def main() -> None:
     """
     Entry point of the bot, should be ran with Docker
     """
+    token = Settings().subscriby_telegram_token
+    bot = Bot(token=token, parse_mode="html")
+    dp = Dispatcher(storage=MemoryStorage())
+
+    dp.message.outer_middleware(RoleMiddleware(admin_list=Settings().subscriby_telegram_admin_ids))
+    dp.callback_query.outer_middleware(RoleMiddleware(admin_list=Settings().subscriby_telegram_admin_ids))
```

Comment:
May be better when there is one instance of the middleware

Reply:
aiogram 3 has a new middlewares that applied for messages and callbacks separately.
We can left only message middlewares, but if we add callbacks for this bot we'll add callback middleware again